packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -28,20 +28,33 @@ const { timeToRead } = require(`./utils/time-to-read`)
 let fileNodes
 let pluginsCacheStr = ``
 let pathPrefixCacheStr = ``
-const astCacheKey = node =>
-  `transformer-remark-markdown-ast-${node.internal.contentDigest}-${pluginsCacheStr}-${pathPrefixCacheStr}`
-const htmlCacheKey = node =>
-  `transformer-remark-markdown-html-${node.internal.contentDigest}-${pluginsCacheStr}-${pathPrefixCacheStr}`
-const htmlAstCacheKey = node =>
-  `transformer-remark-markdown-html-ast-${node.internal.contentDigest}-${pluginsCacheStr}-${pathPrefixCacheStr}`
-const headingsCacheKey = node =>
-  `transformer-remark-markdown-headings-${node.internal.contentDigest}-${pluginsCacheStr}-${pathPrefixCacheStr}`
-const tableOfContentsCacheKey = (node, appliedTocOptions) =>
-  `transformer-remark-markdown-toc-${
-    node.internal.contentDigest
-  }-${pluginsCacheStr}-${JSON.stringify(
-    appliedTocOptions
-  )}-${pathPrefixCacheStr}`
+
+const CACHE_TYPE_AST = `ast`
+const CACHE_TYPE_HTMLAST = `html-ast`
+const CACHE_TYPE_HTML = `html`
+const CACHE_TYPE_HEADINGS = `headings`
+const CACHE_TYPE_TOC = `toc`
+const CACHE_TYPE_NODE_DEPS = `node-dependencies`
+const CACHE_TYPE_NODE_DEP_DIGEST = `node-dependency-digest`
+
+const cacheTypesWithOptions = [CACHE_TYPE_TOC]
+const cacheTypesUsingNodeId = [CACHE_TYPE_NODE_DEP_DIGEST]
+
+const getCacheKey = (cacheType, node, options = {}) => {
+  const keyVars = [
+    cacheTypesUsingNodeId.indexOf(cacheType) >= 0
+      ? node.id
+      : node.internal.contentDigest,
+    pluginsCacheStr,
+    pathPrefixCacheStr,
+  ]
+
+  if (cacheTypesWithOptions.indexOf(cacheType) >= 0) {
+    keyVars.push(JSON.stringify(options))
```

Comment:
Previously, the stringified options were between the `pluginsCacheStr` and the `pathPrefixCacheStr`. I wanted to `push` this to the end just for cleanliness' sake, as opposed to `splice`-ing it in. 

Will this cause longer build first times for projects after they upgrade? I would like to leave this as a `push`, but can change if needed to avoid invalidating valid caches.

Reply:
Since this would come with a package update, we'd invalidate the entire cache anyways. Not an issue! :partying_face: